dokuwiki-text/src/main/java/org/xwiki/contrib/dokuwiki/text/internal/input/DokuWikiInputFilterStream.java

```diff
@@ -282,29 +282,34 @@ private void readDocument(File directory, String dokuwikiDataDirectory, DokuWiki
-                            && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_MODIFIED))
-                            && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_CREATED)
-                            < documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_MODIFIED))
-                    {
-                        //Wiki Document Locale
-                        proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
-                        //read revisions
-                        readPageRevision(file, dokuwikiDataDirectory, proxyFilter);
-                    } else {
-
-                        String pageContents = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
-                        String convertedContent = parseContent(pageContents);
-                        documentLocaleParameters.put(WikiDocumentFilter.PARAMETER_CONTENT, convertedContent);
-                        //Wiki Document Locale
-                        proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
-                        readAttachments(pageContents, file, dokuwikiDataDirectory, proxyFilter);
+                    File f = new File(fileMetadataPath);
+                    if(f.exists() && !f.isDirectory()) {
+                        String metadataFileContents = FileUtils.readFileToString
+                                (new File(fileMetadataPath), "UTF-8");
+                        MixedArray documentMetadata = Pherialize.unserialize(metadataFileContents).toArray();
+                        readDocumentParametersFromMetadata(documentMetadata, documentLocaleParameters);
+
+                        //Wiki document revision
+
+                        if ((documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_CREATED)
+                                && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_MODIFIED))
+                                && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_CREATED)
+                                < documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_MODIFIED))
+                        {
+                            //Wiki Document Locale
+                            proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
+                            //read revisions
+                            readPageRevision(file, dokuwikiDataDirectory, proxyFilter);
+                        } else {
+                            documentLocaleParameters = readDocument(file,documentLocaleParameters,
+                                    dokuwikiDataDirectory, proxyFilter);
+                        }
+                    }
+                    else {
+                        this.logger.warn("File [{}] not found (Ignoring)", f);
```

Comment:
You skipped part of my previous comment that was insinuating that "Ignoring" is a bit light and it would be better to shortly explaining what are the possible consequences of this missing file :)

Reply:
@nehagup This would help the user understand the impact of the file being missing (without really knowing about the technical details about dokuwiki internal file structure). You could warn about the details that we pick from the .meta files, and also maybe attach [this link](https://www.dokuwiki.org/devel:metadata) in the warning message.